src/robotFactory.js

```diff
@@ -1,17 +1,74 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
+BaseRobot.prototype = {
+  goForward(y = 1) {
+    this.coords.y += y;
+  },
+  goBack(y = 1) {
+    this.coords.y -= y;
+  },
+  goRight(x = 1) {
+    this.coords.x += x;
+  },
+  goLeft(x = 1) {
+    this.coords.x -= x;
+  },
+  getInfo() {
+    return 'Robot: ' + this.name
+    + ', Chip version: ' + this.chipVersion
+    + ', Weight: ' + this.weight;
+  },
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
+  this.coords.z = 0;
 }
 
-function DeliveryDrone() {
-  // implement
+FlyingRobot.prototype = {
+  goUp(z = 1) {
+    this.coords.z += z;
+  },
+  goDown(z = 1) {
+    this.coords.z -= z;
+  },
+};
+
+Object.setPrototypeOf(FlyingRobot.prototype, BaseRobot.prototype);
+
+function DeliveryDrone(name, weight, coords,
+  chipVersion, maxLoadWeight, currentLoad) {
```

Comment:
Якщо переносиш параметри, то перенось всі в новий рядок
```suggestion
function DeliveryDrone(
  name,
  weight,
  coords,
  chipVersion,
  maxLoadWeight,
  currentLoad
) {
```